src/middlewares/authentication-middleware.ts

```diff
@@ -1,22 +1,36 @@
-import { NextFunction, Request, Response } from 'express'
+import { Method, ZodiosPathsByMethod } from '@zodios/core'
+import { ZodiosRequestHandler } from '@zodios/express'
+import { Context } from '../config/zodios'
+import { Api } from '../config/zodios/api-schema'
+
+type Middleware = ZodiosRequestHandler<
+	Api,
+	Context,
+	Method,
+	ZodiosPathsByMethod<Api, Method>
+>
```

Comment:
This is a generic Middleware definition and should allow you to plug it to any api definition